lib/llvmopencl/TargetAddressSpaces.cc

```diff
@@ -565,6 +565,11 @@ TargetAddressSpaces::runOnModule(llvm::Module &M) {
     addrSpaceMapDown[POCL_AS_FAKE_GLOBAL] = 1;
     addrSpaceMapDown[POCL_AS_FAKE_LOCAL] = 3;
     addrSpaceMapDown[POCL_AS_FAKE_CONSTANT] = 2;
+  } else if (arch.startswith("nvptx")) {
```

Comment:
Yes, it definitely works without fake address spaces now since we require LLVM 4.0. I wasn't sure what to do with this code - should we leave it there, just remove it, or assert that we shouldn't get here at all if we're using the PTX backend?